### PR TITLE
More Python 3.12 enablement (home-assistant edition)

### PR DESCRIPTION
**Pull request by mweinelt**
## Description of changes

Reverting the pbr disable for 3.12 unblocks lots of packages, which of course need lots of attention. :shrug: 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
Instead of writing comments for all my nits and the switches to pynose and putting the work on you, I've opted into doing it myself and pushed it onto the branch. I've built all the packages locally on this branch and verified that things still work. The diff can be found here https://github.com/NixOS/nixpkgs/compare/03da39652278fa3168d326c5be28c8302b493064..9d42fd66e470ecf2b1d45ac904dae68de1916bd0
I hope that's fine for you.
Please consider everything up to this point reviewed and approved by me.

File: pkgs/development/python-modules/pytest-benchmark/default.nix [hunk 1]
```diff
@@ -66,6 +66,10 @@ buildPythonPackage rec {
     pytest-xdist
   ] ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
 
+  pytestFlagsArray = [
+    "-W" "ignore::DeprecationWarning"
```
[the remark this answers is not in the document]
Comment (mweinelt) [reply pkgs/development/python-modules/pytest-benchmark/default.nix:70]: Feel free to send a PR.

**Comment by mweinelt** [comment]
Extracted the opencv bump into #289994 for more review.

**Comment by mweinelt** [comment]
```
❯ ./maintainers/scripts/rebuild-amount.sh origin/staging-next
Estimating rebuild amount by counting changed Hydra jobs (parallel=unset).
      1 pkgs-lib-tests
  13805 x86_64-darwin
  23454 x86_64-linux

**Comment by infinisil** [comment]
Fyi, I believe the `pkgs/by-name` check failed due to the base branch change, probably staging-next didn't have https://github.com/NixOS/nixpkgs/pull/282707 yet or so. Shouldn't occur again in the future. Please ping me if you see the check failing though, it generally shouldn't be ignored, I'm continuously improving it.

**Pull request merged**